tests/extras/datasets/pandas/test_sql_dataset.py

```diff
@@ -224,12 +233,71 @@ def test_table_exists_mocked(self, mocker, table_data_set):
         self._assert_sqlalchemy_called_once()
 
 
-class TestSQLQueryDataSet:
-    @staticmethod
-    def _assert_pd_called_once():
-        _callable = pd.read_sql_query
-        _callable.assert_called_once_with(sql=SQL_QUERY, con=CONNECTION)
+class TestSQLTableDataSetSingleConnection:
+    def test_single_connection_performance(self, dummy_dataframe):
+        kwargs = dict(
+            save_args=dict(if_exists="append"),
+            table_name=TABLE_NAME,
+            credentials=dict(con=CONNECTION),
+        )
+        datasets = [SQLTableDataSet(**kwargs) for _ in range(10)]
+
+        for d in datasets:
+            d.save(dummy_dataframe)
+
+        start = time.time()
+        for d in datasets:
+            d.load()
+        end = time.time()
+
+        delta = end - start
+        print(delta)
+        cons = datasets[0].engines
+        for con in cons.values():
+            con.dispose()
+
+    def test_single_connection(self, dummy_dataframe, mocker):
+        """Test to make sure multiple instances use the same connection object."""
+        dummy_to_sql = mocker.patch.object(dummy_dataframe, "to_sql")
+        kwargs = dict(table_name=TABLE_NAME, credentials=dict(con=CONNECTION))
+
+        first = SQLTableDataSet(**kwargs)
+        unique_connection = first.engines[CONNECTION]
+        datasets = [SQLTableDataSet(**kwargs) for _ in range(10)]
+
+        for ds in datasets:
+            ds.save(dummy_dataframe)
+            engine = ds.engines[CONNECTION]
+            assert engine is unique_connection
+
+        expected_call = mocker.call(name=TABLE_NAME, con=unique_connection, index=False)
+        dummy_to_sql.assert_has_calls([expected_call] * 10)
+
+        for ds in datasets:
+            ds.load()
+            engine = ds.engines[CONNECTION]
+            assert engine is unique_connection
+
+    def test_create_connection_only_once(self, mocker):
+        """Test that two datasets that need to connect to the same db
+        only create a connection once.
+        """
+        mock_engine = mocker.patch(
+            "kedro.extras.datasets.pandas.sql_dataset.create_engine"
+        )
+        first = SQLTableDataSet(table_name=TABLE_NAME, credentials=dict(con=CONNECTION))
+        assert len(first.engines) == 1
+
+        second = SQLTableDataSet(
+            table_name="other_table", credentials=dict(con=CONNECTION)
+        )
+        assert len(second.engines) == 1
+        assert len(first.engines) == 1
+
+        mock_engine.assert_called_once_with(CONNECTION)
```

Comment:
Do we need a test to check that two different connection strings create two different engines?